internal/node-daemon/daemon_controller.go

```diff
@@ -0,0 +1,31 @@
+package nodedaemon
```

Comment:
I think it's better to be consistent with naming. I see the folder is named node-daemon whereas, under cmd, we have nodedaemon/main.go.